### DISCUSSION TITLE
Extending USB discovery to non-serial devices

### DISCUSSION THREAD

**Discussion by scop** [home-assistant/architecture#748] [General] [answered]
I'm wondering if there's a particular reason we are limiting USB discovery to serial/tty devices only?

I had a brief look if implementing USB discovery for Huawei LTE would be feasible, but the current limitation prevents discovery events for those kinds of devices. (Aside, the experience with this kind of detection wouldn't be the smoothest possible as a USB discovery event carries no info about the IP address we'd need to access the device, but I suppose it could be seen useful nevertheless.)

I've some local PoC code to generate those events for all kinds of devices that I could look into beating into shape if this is something we'd like to see done. Mainly the on demand/startup tree scan part is missing, live added devices are taken care of already through the udev events.

**Comment by bdraco** [comment, marked as answer]
> I'm wondering if there's a particular reason we are limiting USB discovery to serial/tty devices only?
> 
There was no specific intent to limit to serial/tty devices only.  As long as the device has a repeatable unique id and the implementation doesn't generate significant additional load on the host, it seems reasonable to extend it.
